presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -240,6 +240,12 @@ public CostProvider getCostProvider()
             {
                 return costProvider;
             }
+
+            @Override
+            public long getTimeoutInMillis()
+            {
+                return context.timeoutInMilliseconds;
```

Comment:
This has no meaning from Rules' perspective. It would be better to expose `boolean isTimeLimitExhausted()` instead.

Reply:
> a void check that throws an exception

sounds even better than what i proposed